app/src/main/java/com/example/vmedvediev/redditapp/presenter/LoginPresenter.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.vmedvediev.redditapp.presenter
+
+import android.text.TextUtils
+import android.util.Log
+import com.example.vmedvediev.redditapp.model.NetworkManager
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.launch
+import retrofit2.converter.gson.GsonConverterFactory
+
+class LoginPresenter(private val view: LoginPresenter.View) {
+
+    companion object {
+        private const val TAG = "PostsPresenter"
+    }
+
+    //This function should be named "login" because its name used in the request.
+    fun login(username: String, password: String) = launch(UI) {
+        if (validateCredentials(username, password)) {
+            view.showLoading()
```

Comment:
Code inside if can be extracted to a separate method, e.g. `startLogin()` or `makeLogin()`